pkg/kube/log_reader.go

```diff
@@ -0,0 +1,41 @@
+package kube
+
+import (
+	"io"
+
+	"k8s.io/client-go/rest"
+)
+
+var _ io.ReadCloser = (*logReader)(nil)
+
+func newLogReader(rw rest.ResponseWrapper) *logReader {
+	return &logReader{rw: rw}
+}
+
+// logReader defers a call to ResponseWrapper.Stream() until Read is called.
+// This is to help handle kubernetes behavior where calls to
+// Pod.GetLogs(...).Stream() will hang until at least one byte is returned.
+// kubectl logs handles this in
+// https://github.com/kubernetes/kubernetes/pull/67573/files#diff-12d472fe036bbe778e84dffc71564eb1R355
+type logReader struct {
+	rw  rest.ResponseWrapper
+	err error
```

Comment:
What is `err` used for?

Reply:
NM, saw it

Reply:
this could be used in the Read method. Let me push that down